gettsim/benefits/arbeitsl_geld_2.py

```diff
@@ -46,131 +59,92 @@ def alg2(household, params):
+    kinder_satz_m = 0
+    for age_lims in [(0, 6), (7, 13), (14, 24)]:
+        kinder_satz_m += (
+            params["regelsatz"]
+            * params["anteil_regelsatz"][f"kinder_{age_lims[0]}_{age_lims[1]}"]
+            * (
+                household["kind"] & household["alter"].between(age_lims[0], age_lims[1])
+            ).sum()
+        )
 
-    household["regelsatz_m"] = params["calc_regelsatz"](
-        household, children_age_info, params
-    )
+    if num_adults == 1:
+        household["regelsatz_m"] = params["regelsatz"] * (1 + household["mehrbed"])
+    elif num_adults > 1:
+        household["regelsatz_m"] = params["regelsatz"] * (
+            params["anteil_regelsatz"]["zwei_erwachsene"] * (2 + household["mehrbed"])
+            + (
+                params["anteil_regelsatz"]["weitere_erwachsene"]
+                * np.maximum((num_adults - 2), 0)
+            )
+        )
+    household["regelsatz_m"] += kinder_satz_m
 
     return household
 
 
-def regelberechnung_until_2010(household, children_age_info, params):
-    if children_age_info["anz_erw"] == 1:
-        return (
-            params["rs_hhvor"] * (1 + household["mehrbed"])
-            + (
-                params["rs_hhvor"]
-                * params["a2ch14"]
-                * children_age_info["child14_24_num"]
-            )
-            + (
-                params["rs_hhvor"]
-                * params["a2ch7"]
-                * children_age_info["child7_13_num"]
-            )
-            + (
-                params["rs_hhvor"]
-                * params["a2ch0"]
-                * (
-                    children_age_info["child0_2_num"]
-                    + children_age_info["child3_6_num"]
-                )
-            )
-        )
-    elif children_age_info["anz_erw"] > 1:
-        return (
-            (
-                params["rs_hhvor"] * params["a2part"] * (1 + household["mehrbed"])
-                + (params["rs_hhvor"] * params["a2part"])
-                + (
-                    params["rs_hhvor"]
-                    * params["a2ch18"]
-                    * np.maximum((children_age_info["anz_erw"] - 2), 0)
-                )
-            )
-            + (
-                params["rs_hhvor"]
-                * params["a2ch14"]
-                * children_age_info["child14_24_num"]
-            )
-            + (
-                params["rs_hhvor"]
-                * params["a2ch7"]
-                * children_age_info["child7_13_num"]
-            )
-            + (
-                params["rs_hhvor"]
-                * params["a2ch0"]
-                * (
-                    children_age_info["child0_2_num"]
-                    + children_age_info["child3_6_num"]
-                )
-            )
+def regelberechnung_2011_and_beyond(household, params):
+
+    num_adults = len(household) - household["kind"].sum()
+    kinder_satz_m = 0
+    # For kids we count the "Regelbedarfstufen" from 6 to 4 downwards.
```

Comment:
It should not be "we count" but "Regelbedarfsstufen are ordered inversely age of the child"

I would prefer if we went through the Regelbedarfsstufen from 1-6, though. Makes it easier to think through.

Reply:
Changed the order and comment. Now it should be more clearer!